src/utils/getImageDataFromURL.ts

```diff
@@ -1,15 +1,69 @@
 import sha1 from 'js-sha1';
-import { getSketchVersion } from './getSketchVersion';
-import sketchMethod from '../jsonUtils/sketchImpl/makeImageDataFromUrl';
-import nodeMethod from '../jsonUtils/nodeImpl/makeImageDataFromUrl';
+import { PlatformBridge } from '../types';
 
-const makeImageDataFromUrl = (url?: string): { data: string; sha1: string } => {
-  const data = getSketchVersion() === 'NodeJS' ? nodeMethod(url) : sketchMethod(url);
+const ERROR_IMAGE =
+  'iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mM8w8DwHwAEOQHNmnaaOAAAAABJRU5ErkJggg==';
 
-  return {
-    data,
-    sha1: sha1(data),
-  };
+const ERROR_RESULT = {
+  data: ERROR_IMAGE,
+  sha1: sha1(ERROR_IMAGE),
 };
 
-export default makeImageDataFromUrl;
+const isImage = (buffer: Buffer): boolean => {
+  const firstByte = buffer[0];
+
+  // Check for first byte to see if we have an image.
+  // 0xFF = JPEG, 0x89 = PNG, 0x47 = GIF, 0x49 = TIFF, 0x4D = TIFF
+  return (
+    firstByte === 0xff ||
+    firstByte === 0x89 ||
+    firstByte === 0x47 ||
+    firstByte === 0x49 ||
+    firstByte === 0x4d
+  );
+};
+
+const fetchRemoteImage = async (url: string, { fetch }: PlatformBridge): Promise<Buffer> => {
+  const response = await fetch(url);
+  if (!response.ok) {
+    throw new Error(`${response.status}`);
+  }
+
+  const arrayBuffer = await response.arrayBuffer();
+  if (Buffer.isBuffer(arrayBuffer)) {
+    // skpm polyfill returns a Buffer instead of an ArrayBuffer
+    return arrayBuffer;
```

Comment:
let's put that in the bridge maybe?

Reply:
I'm not sure that this one is worth making an API for in the bridge, because in theory every implementation should be returning an `ArrayBuffer` and it's just an idiosyncrasy of the skpm one that it returns a `Buffer` directly instead.

What I could do here though is removing this check and using always `Buffer.from(arrayBuffer)`. For all the compliant implementations it will work as expected. For the skpm one, it will just copy the memory again, that doesn't do much harm; and if you feel like patching the polyfill at a later date, it would still work fine and become a little bit more efficient. What do you think?